src/Enum.php

```diff
@@ -61,14 +61,24 @@ public function __construct($value)
             $value = $value->getValue();
         }
 
-        if (!$this->isValid($value)) {
-            throw new \UnexpectedValueException("Value '$value' is not part of the enum " . static::class);
-        }
+        static::assertValidValue($value);
 
         /** @psalm-var T */
         $this->value = $value;
     }
 
+    /**
+     * @param mixed $value
+     * @return static
+     * @psalm-return static<T>
+     */
+    public static function fromMixed($value): self
```

Comment:
Not a bad thing to create the Enum instance by value.
Considering that this library will need to have a version that will be migrated easily to normal PHP enum that will arrive in 8.1: https://github.com/myclabs/php-enum.git
The method would be named `from`.

Reply:
Means suggest to rename the named ctor to `from`?

Reply:
Sorry, I shared the wrong link.
I was suppose to be this one: https://wiki.php.net/rfc/enumerations

Reply:
Changed and added an additional test for named ctor.

Reply:
Ohhh very good point, thanks for those changes!

I used to be skeptical about that PR, but now I'm sold!